packages/client/hmi-client/src/components/responsive-matrix/matrix.vue

```diff
@@ -0,0 +1,902 @@
+	// components                                                                   //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	components: {
+		// ResponsiveCellBarContainer,
+		// ResponsiveCellLineContainer
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// props                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	props: {
+		data: {
+			type: null,
+			default: [[], []] // e.g. [[{}, {}, {}], [{}, {}, {}]]
+		},
+		// labels: {
+		// 	type: null,
+		// 	default: [[], []],
+		// },
+		cellLabelRow: {
+			type: Array as PropType<number[] | string[]>,
+			default() {
+				return [];
+			}
+		},
+		cellLabelCol: {
+			type: Array as PropType<number[] | string[]>,
+			default() {
+				return [];
+			}
+		},
+		fillColorFn: {
+			type: Function,
+			default() {
+				return '#000000';
+			}
+		},
+		barColorFn: {
+			type: Function,
+			default() {
+				return '#000000';
+			}
+		},
+		lineColorFn: {
+			type: Function,
+			default() {
+				return '#000000';
+			}
+		},
+		backgroundColor: {
+			type: String,
+			default: '#FFFFFF'
+		},
+		cellMarginRow: {
+			type: [String, Array],
+			default: '2px'
+		},
+		cellMarginCol: {
+			type: [String, Array],
+			default: '2px'
+		}
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// data                                                                         //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	data() {
+		return {
+			dataCellList: [] as object[],
+			dataRowList: [] as object[][], // e.g. [[row1col1Obj, row1col2Obj]]
+			dataColList: [] as object[][], // e.g. [[row1col1Obj, row2col1Obj]]
+			dataParameters: new Set() as Set<string>, // e.g. ["age", "height", "weight"]
+			dataParametersMin: {}, // e.g. {param1: 0, param2: 3}
+			dataParametersMax: {}, // e.g. {param1: 10, param2: 17}
+			labelRowList: [] as number[] | string[],
+			labelColList: [] as number[] | string[],
+			numRows: 0,
+			numCols: 0,
+
+			microRowArray: new Uint32Array(0) as Uint32Array,
+			microColArray: new Uint32Array(0) as Uint32Array,
+
+			uniforms: {} as Uniforms,
+			worldWidth: 0,
+			worldHeight: 0,
+
+			screenHeight: 0,
+			screenWidth: 0,
+
+			selectedCellList: [] as SelectedCell[], // e.g. [[startrow, startcol, endrow, endcol]]
+			selectedRows: [] as CellStatus[], // e.g. [0, 0, 1, 0]
+			selectedCols: [] as CellStatus[], // e.g. [0, 0, 1, 0]
+			selectedCell: Array(4) as SelectedCell, // e.g. [startrow, startcol, endrow, endcol]
+
+			update: 0,
+			move: 0,
```

Comment:
Not too clear what move/update are doing, can you add a comment to clarify?

Reply:
yup, they exist basically to trigger updates in children components when either is required, this should probably have been commented out here and uncommented in the drilldown pr. will add comment